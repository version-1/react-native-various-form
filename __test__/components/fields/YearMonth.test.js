import React, { Component } from 'react'
import { Text, StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import { YearMonth } from 'react-native-various-form'

describe('YearMonth', () => {
  it('default', () => {
    const tree = renderer.create(<YearMonth value="2018/10" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('LongYearMonth', () => {
    const tree = renderer
      .create(<YearMonth formatType="longYearMonth" value="2018/10" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('ShortYearMonth', () => {
    const tree = renderer
      .create(<YearMonth formatType="shortYearMonth" value="2018/10" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('LongMonthYear', () => {
    const tree = renderer
      .create(<YearMonth formatType="longMonthYear" value="2018/10" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('ShortMonthYear', () => {
    const tree = renderer
      .create(<YearMonth formatType="shortMonthYear" value="2018/10" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
