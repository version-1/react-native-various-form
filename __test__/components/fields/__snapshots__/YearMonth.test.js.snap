// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearMonth LongMonthYear 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="longMonthYear"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="MM"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="10"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
          },
          undefined,
        ]
      }
    >
      /
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="longMonthYear"
      keyboardType="numeric"
      maxLength={4}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="YYYY"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="2018"
    />
  </View>
</View>
`;

exports[`YearMonth LongYearMonth 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="longYearMonth"
      keyboardType="numeric"
      maxLength={4}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="YYYY"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="2018"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
          },
          undefined,
        ]
      }
    >
      /
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="longYearMonth"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="MM"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="10"
    />
  </View>
</View>
`;

exports[`YearMonth ShortMonthYear 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="shortMonthYear"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="MM"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="10"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
          },
          undefined,
        ]
      }
    >
      /
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="shortMonthYear"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="YY"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="18"
    />
  </View>
</View>
`;

exports[`YearMonth ShortYearMonth 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="shortYearMonth"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="YY"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="18"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
          },
          undefined,
        ]
      }
    >
      /
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      formatType="shortYearMonth"
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="MM"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="10"
    />
  </View>
</View>
`;

exports[`YearMonth default 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      maxLength={2}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="MM"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="10"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
          },
          undefined,
        ]
      }
    >
      /
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      maxLength={4}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="YYYY"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
      value="2018"
    />
  </View>
</View>
`;
